### ISSUE TITLE
Feature request: Add commands to `editor/context` contribute submenu. 

### ISSUE THREAD

**Issue by peaceshi** [yzhang-gh/vscode-markdown#1358]
### Proposal

From https://github.com/mdickin/vscode-markdown-shortcuts/issues/76 . 
This issue has not been replied to in a month and seems like the project has been no one maintains.
So, I want to add it to markdown-all-in-one.

I will make a PR later.

### Other information

![image](https://github.com/yzhang-gh/vscode-markdown/assets/10380828/eae67736-4a10-4303-80cc-f7fb55c3527b)

**Comment by yzhang-gh** [comment]
Thanks for your contribution.

However, I don't think it is a good UX design. For exporting as HTML, it requires many key strokes (ctrl shift p, p, r, i, n, t, (optional arrow keys), enter) and thus it will be convenient to have a menu item. For styling commands that is not true, it is just ctrl b/i etc. And it is even worse when you need to go into a submenu.

**Comment by peaceshi** [comment]
It is not that everyone is a programmer or pro. It is difficult for civilian personnel to remember the right key bind then they will check it again and again. How about just adding them to the menu instead of submenu?

**Comment by yzhang-gh** [comment]
> How about just adding them to the menu instead of submenu?

Then there will be too many commands.

The keybindings are so widely used (just look the **B** and _I_ above the GitHub comment box or other software like Word). I am not very convinced of the necessity. We can leave this open and see what other users think.

**Comment by peaceshi** [comment]
How about adding to the title bar?
![image](https://github.com/yzhang-gh/vscode-markdown/assets/10380828/6751d772-8a5a-4e1e-b7df-d7c93069b21f)

or like this?
![image](https://github.com/yzhang-gh/vscode-markdown/assets/10380828/115b8f65-3433-4491-9f35-2e47a76d99cb)

**Comment by MomentDerek** [comment]
I support adding it to the menu! It is useful because I like to click it directly.

**Comment by LiXiaYu** [comment]
> > How about just adding them to the menu instead of submenu?
> 
> Then there will be too many commands.
> 
> The keybindings are so widely used (just look the **B** and _I_ above the GitHub comment box or other software like Word). I am not very convinced of the necessity. We can leave this open and see what other users think.

You're right, but the noble VScode supports the mouse and has multi-level menu options that can be freely switched on and off.

This is not Vim, we don't need comment box!

**Comment by lhmouse** [comment]
Please consider 'non-well-known' key bindings bad.

So, what are 'well-known' keybindings? For example, Ctrl-C copies the selection; okay good, we are used to it. Ctrl-Insert also copies, and sometimes it's essential such as in the terminal, where Ctrl-C would send an interrupt; so, good, too.

They're all good, until there are _too many_. So what does Ctrl-Shift-R do? No, nobody will bother remembering that. It does not have to exist. Please give my menus back.

N.B. The Ctrl-B combination conflicts with `workbench.action.toggleSidebarVisibility`, so it's already not good; but not to me, as Ctrl-B doesn't work in the terminal anyway, so I have Ctrl-Shift-B to close it.

**Comment by yzhang-gh** [comment]
> How about adding to the title bar?
>
> ![image](https://private-user-images.githubusercontent.com/10380828/287948222-6751d772-8a5a-4e1e-b7df-d7c93069b21f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE4NjY4NzQsIm5iZiI6MTcwMTg2NjU3NCwicGF0aCI6Ii8xMDM4MDgyOC8yODc5NDgyMjItNjc1MWQ3NzItOGE1YS00ZTFlLWI3ZGYtZDdjOTMwNjliMjFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA2VDEyNDI1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2M2M0N2VlOTMxOWYyMTBjM2E2ZWM5NTI2ZjY1YmM5NDFlODEyZTBmNzZiYWJlMzg4YzRlZWYwNzliZmQxM2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hVOeJnNXfQubb6xGypFqOvcxvjPH-sKHwnLjdzARvtE)

This looks much better to me, not only because it is more convenient than context menu, but also because it shares the same UX as other editors.

VS Code has [built-in icons](https://code.visualstudio.com/api/references/icons-in-labels), or you can find them [here](https://github.com/microsoft/vscode-codicons).

---

> N.B. The Ctrl-B combination conflicts with `workbench.action.toggleSidebarVisibility`, so it's already not good; but not to me, as Ctrl-B doesn't work in the terminal anyway, so I have Ctrl-Shift-B to close it.

I see. (related #89)

**Comment by yzhang-gh** [comment]
Just FYI, I have added an option `showActionButtons` to show/hide these button. This is `false` by default following the [VS Code UX guidelines](https://code.visualstudio.com/api/ux-guidelines/editor-actions).

If you would like to check it out, you can install the latest dev build
https://github.com/yzhang-gh/vscode-markdown/actions/runs/7322269176/artifacts/1134210918

I am going to release a new stable version soon if everything works well.

**Comment by peaceshi** [comment]
> Just FYI, I have added an option `showActionButtons` to show/hide these button. This is `false` by default following the [VS Code UX guidelines](https://code.visualstudio.com/api/ux-guidelines/editor-actions).
> 
> If you would like to check it out, you can install the latest dev build https://github.com/yzhang-gh/vscode-markdown/actions/runs/7322269176/artifacts/1134210918
> 
> I am going to release a new stable version soon if everything works well.


分组应该跟之前一样在同一组里，否则如果有多个需要在工具栏显示操作的插件就会被打乱顺序。而且vsc只显示前几个注册的，后面的会被放到下拉菜单里。
![image](https://github.com/yzhang-gh/vscode-markdown/assets/10380828/ff2f18ff-d06d-49d3-bcfb-d796303f082a)
![image](https://github.com/yzhang-gh/vscode-markdown/assets/10380828/1bebfa10-cba8-4b07-94b3-c3d6eec6847e)

**Comment by yzhang-gh** [comment]
因为我观察到之前的 `1_1`，`1_2` 写法并不能控制顺序，其实是按照命令的字母顺序排列的，比如 code span 会在 italic 之前。

考虑到现在的问题，那需要更了解 VS Code 这里的排序逻辑了

**Comment by ulville** [comment]
How do I enable this? I could not find any option called `showActionButtons`

**Comment by yzhang-gh** [comment]
Thanks for the feedback. There is a mistake and it is not shown up in the Settings GUI. You can directly put `"markdown.extension.showActionButtons": true` in your settings JSON. Or you can install the dev build here
https://github.com/yzhang-gh/vscode-markdown/actions/runs/7541713779/artifacts/1172167689
